### ISSUE TITLE
Layout update broken, no eq-lines in windows, names in colour no longer editable

### ISSUE THREAD

**Issue by ffAudio** [ffAudio/foleys_gui_magic#25]
_Originally posted by @Werner88 in https://github.com/ffAudio/foleys_gui_magic/pull/24#issuecomment-595689970_

Hi Daniel,

thank you very much for this framework, I like it very much!!!

I have problems since your last update:

1)
a) after loading xml file, same after the first launch, the layout looks horrible
b) after changing the lookandfeel of a component (e.g. a slider) the layout looks horrible
after resizing everything is okay.

Maybe it has something to do with a missing call to 
builder->updateLayout(); 

2)
In windows the EqualizerExample is broken, no eq-lines are drawn.

3)
it is never possible to use the color names because whenever you click into a color field you are in the colourSelector.

I implemented this feature myself in a different way.
For the color fields I set the buttonText to „o“. Only after clicking the button the colourSelector launches.
The color field stays editable. When making the color field empty, the „0“ turns to an „x“. After clicking the „x“ you get the node.removeProperty call.
Anyway, it would be nice to have the color fields editable like before.

kind regards
and again thank you for guiMagic

Werner

**Comment by ffAudio** [comment]
Hi Werner,

Thanks for the feedback! That's much appreciated!

Yes, I realised my mistake. By optimising away the recreation of the components the missing layout update became appearant. I created a fix yesterday on the develop branch and will merge it right away.

I will have to look into that. As a one man team I forgot to test the windows. I only have a smoke test in my automatic builds.

Which eq-lines do you mean? The draggable lines or the frequency response?
Does it happen with a new plugin instance as well? Note that changes of the GUI are now persistent in that plugin instance. By opening a fresh instance, you will see the appearance, that a user will get.
And if you exported the XML, you still need to re-save the project to update the BinaryData.

Indeed, i sacrificed the colour names. I need a way to bring them back. Unfortunately I don't like the idea of double meaning of the button. That one is meant to reset to the inherited value.
I will improve the colour picker anyway, it should have HSL sliders as well, and colour swatches to make it easier to copy colours over. Maybe a field to type the colour name there would be an option?
I think that feature needs a bit more thought, I am open for suggestions.

Cheers,
Daniel

**Comment by ffAudio** [comment]
As of release 10.04 on 10. March 2020:
- Layout update is fixed, 
- eq-lines were actually working, 
- performance on windows in the example is back to normal by re-adding openGL
- Colours can be typed in again, and the ColourSelector has swatches to save colours for later use
